### PR TITLE
Add create vesting hook UI

### PR DESCRIPTION
**Pull request by JeanNeiverth**
Includes:
- add two new inputs to cow-hooks-ui: PeriodWithScaleInput and TokenAmountInput
- add create vesting UI using these fields
- Fields validations
- env changes to use pnpm
- env changes to fix issues when importing libraries

Currently doesn't include:
- bring real token from swap (using static ETH)

image

image

image

### CODE REVIEW

File: apps/create-vesting/public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  ],
+  "orientation": "portrait",
+  "name": "Create Vesting",
+  "short_name": "Create Vesting",
+  "start_url": ".",
+  "theme_color": "#ffffff",
+  "cow_hook_dapp": {
+    "name": "Create Vesting",
+    "descriptionShort": "Create a LlamaPay vesting contract",
+    "description": "Create Vesting is a powerful hook designed to streamline the process of setting up a LlamaPay vesting contract right after your swap.",
```
Comment (yvesfracari) [inline apps/create-vesting/public/manifest.json:28]: 👍

File: apps/create-vesting/src/app/page.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import { useForm } from "react-hook-form";
+
+import { createVestingSchema, periodScaleOptions } from "#/utils/schema";
+
+//will be used
+const scaleToSecondsMapping = {
+  Day: 24 * 60 * 60,
+  Week: 7 * 24 * 60 * 60,
+  Month: 30 * 24 * 60 * 60,
+};
```
Comment (yvesfracari) [inline apps/create-vesting/src/app/page.tsx:31]: This should be in a utils

File: apps/create-vesting/src/app/page.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+  useEffect(() => {
+    const { actions } = initCoWHookDapp({ onContext: setContext });
+    setActions(actions);
+  }, []);
+
+  useEffect(() => {
+    const newTheme = isDarkMode ? "dark" : "light";
+    document.documentElement.setAttribute("data-theme", newTheme);
+  }, [isDarkMode]);
```
Comment (yvesfracari) [inline apps/create-vesting/src/app/page.tsx:59]: On the next PR we can move this to context I think

File: apps/create-vesting/src/app/page.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      {context && (
+        <Form {...form} className="contents">
+          <Wrapper>
+            <ContentWrapper>
+              <Input
+                name="recipient"
+                label="Recipient"
+                placeholder="0xabc..."
+                autoComplete="off"
+                className="w-full h-12 mt-0 p-2.5 rounded-xl outline-none text-color-text-paper border-2 border-color-border bg-color-paper-darker placeholder:opacity-100"
```
Comment (yvesfracari) [inline apps/create-vesting/src/app/page.tsx:94]: There is a lot of stylling already with bleu/ui component. Are the styling work? 🤔

File: apps/create-vesting/src/app/page.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+              <br />
+              <div className="flex gap-4 w-full">
+                <PeriodWithScaleInput
+                  periodScaleOptions={periodScaleOptions}
+                  namePeriodValue="period"
+                  namePeriodScale="periodScale"
+                  label="Period"
+                  validation={{ valueAsNumber: true, required: true }}
+                  onKeyDown={(e) =>
+                    ["e", "E", "+", "-"].includes(e.key) && e.preventDefault()
```
Comment (yvesfracari) [inline apps/create-vesting/src/app/page.tsx:105]: why we need this?

File: apps/create-vesting/src/global.css [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+  --accent-foreground: var(--cow-color-white);
+  --popover: var(--cow-color-paper);
+  --popover-foreground: var(--cow-color-text);
+  --card: var(--cow-color-paper);
+  --card-foreground: var(--cow-color-text);
+  --radius: var(--cow-border-radius-normal);
+  --highlight: var(--cow-color-info);
+  --info: var(--cow-color-info);
+  --success: var(--cow-color-success);
+  --warning: var(--cow-color-warning);
```
Comment (yvesfracari) [inline apps/create-vesting/src/global.css:81]: 👌

File: apps/create-vesting/src/utils/schema.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  recipient: z
+    .string()
+    .min(1, "Address is required")
+    .refine(isAddress, "Insert a valid Ethereum address"),
+  period: z
+    .number({ message: "Invalid amount" })
+    .gt(0, "Period must be greater than 0"),
+  periodScale: z
+    .string()
+    .refine(refinePeriodScale, "Scale must be one of the options"),
```
Comment (yvesfracari) [inline apps/create-vesting/src/utils/schema.ts:20]: We could use an enum here instead of string

File: apps/create-vesting/tailwind.config.js [hunk 8]
```diff
@@ -0,0 +1,144 @@
+/* eslint-env node */
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+/** @type {import("tailwindcss").Config} */
+module.exports = {
+  content: [
+    "./src/**/*.{js,ts,jsx,tsx}",
+    "./node_modules/@bleu/ui/dist/**/*",
```
Comment (yvesfracari) [inline apps/create-vesting/tailwind.config.js:8]:
```suggestion
    "./node_modules/@bleu/ui/dist/**/*",
    "../..//node_modules/@bleu/ui/dist/**/*",
```

This should make bleu/ui styling work 😉

File: packages/cow-hooks-ui/src/ui/PeriodWithScaleInput.tsx [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+import {
+  FieldError,
+  RegisterOptions,
+  useFormContext,
+  useFormState,
+  useWatch,
+} from "react-hook-form";
+
+interface IPeriodWithScaleInput
+  extends Omit<HTMLProps<HTMLInputElement>, "name"> {
```
Comment (yvesfracari) [inline packages/cow-hooks-ui/src/ui/PeriodWithScaleInput.tsx:21]: Why omit name?

File: packages/cow-hooks-ui/src/ui/TokenAmountInput.tsx
Comment (yvesfracari) [inline packages/cow-hooks-ui/src/ui/TokenAmountInput.tsx:1]: What is the difference between this in and a regular input?

**Review by yvesfracari — APPROVED** [review]

File: apps/create-vesting/src/app/page.tsx (hunk 5 above)
Comment (JeanNeiverth) [reply apps/create-vesting/src/app/page.tsx:105]: The type="number" input allows inserting [e, E, +, -] by default, so I thought would be kind of weird to allow that, and I noticed that on other CoW amount inputs they block this too

File: packages/cow-hooks-ui/src/ui/TokenAmountInput.tsx
Comment (JeanNeiverth) [reply packages/cow-hooks-ui/src/ui/TokenAmountInput.tsx:1]: The difference is that this one has a token inside it:

image

I'm going to add a tokenSymbol to it to make more sense

**Pull request merged**